src/pmpo_materialPoints.hpp

```diff
@@ -220,9 +214,14 @@ class MaterialPoints {
       return maxAppID;
     }
 
-//MUTATOR  
+    // MUTATOR  
     template <MaterialPointSlice index> void fillData(double value);//use PS_LAMBDA fill up to 1
     void T2LTracking(Vec2dView dx);
+
+    // INTERNAL DO NOT CALL
```

Comment:
This comment seems redundant as these are private member functions.  Who are we telling not to call these functions?

Reply:
These functions are public. I added this comment because the functions have to be public. We have similar comments in pumi-pic. 

After cursory reading it seems like CUDA functions have to be public due to some compiler reasons. Btw here is the error:

`The enclosing parent function ("createDPS") for an extended __host__ __device__ lambda cannot have private or protected access within its class`

Reply:
Ahh.  Sorry about that; I misread the code in the github interface - likely due to the collapsed/hidden sections.

Do these functions access class data?  If not, suppose we had a struct/class with these helper methods listed as public members.  Could we then create a private instance of this struct/class here and call its methods without hitting the cuda error?

Reply:
I think that would probably work, I will run it by Onkar to see what he thinks.

Reply:
Actually before that. We could do that for these functions, but in the future we might want helper functions that can access class data, in that situation do we want our helper functions in two separate places? We could hope that we don't need helper functions like that. If we want to make it more clear we could prepend the names like _i_createDPS()

Reply:
I'm really not sure where this is heading.  Any approach that prevents the user from calling the helper functions is OK with me.

Note, in C++ leading underscores on function names are not allowed (https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier).

Reply:
OK. I think I have a good solution. I moved all of the helper functions into the .cpp file, so they would not be included with the .hpp file. And I also placed them in an unnamed namespace to stop users from being able to access them.